### ISSUE TITLE
Failure to parse link-local IPv6 address

### ISSUE THREAD

**Issue by zart** [Pylons/pyramid_debugtoolbar#166]
```
Traceback (most recent call last):
  File "C:\Program Files (x86)\Python 2.7\Lib\wsgiref\handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "...\lib\site-packages\pyramid-1.5-py2.7.egg\pyramid\router.py", line 242, in __call__
    response = self.invoke_subrequest(request, use_tweens=True)
  File "...\lib\site-packages\pyramid-1.5-py2.7.egg\pyramid\router.py", line 217, in invoke_subrequest
    response = handle_request(request)
  File "...\lib\site-packages\pyramid_debugtoolbar-2.0.2-py2.7.egg\pyramid_debugtoolbar\toolbar.py", line 158, in toolbar_tween
    or not addr_in(last_proxy_addr, hosts) \
  File "...\lib\site-packages\pyramid_debugtoolbar-2.0.2-py2.7.egg\pyramid_debugtoolbar\utils.py", line 184, in addr_in
    if ipaddr.IPAddress(addr) in ipaddr.IPNetwork(host):
  File "...\lib\site-packages\pyramid_debugtoolbar-2.0.2-py2.7.egg\pyramid_debugtoolbar\ipaddr.py", line 85, in IPAddress
    address)
ValueError: 'fe80::e556:2a1a:91e2:7023%15' does not appear to be an IPv4 or IPv6 address
```

This happens because ipaddr doesn't know about xxxx%interface_or_ifindex syntax for link-local IPv6 address.
I fixed this for myself by stripping %ifindex part in _BaseV6._ip_int_from_string but I'm not sure how this will affect matching of debugtoolbar.hosts.

**Comment by mmerickel** [comment]
When does this happen? I can't find an ipaddress library that doesn't error on this address. I tried the stdlib as well as the popular https://pypi.python.org/pypi/ipaddress backport to no avail, as well as the latest version of google's ipaddr which the toolbar is currently vendoring.

**Comment by bertjwregeer** [comment]
This will happen if you use mDNS to connect to an HTTP server running on a local network machine that happens to have both IPv4/IPv6 connectivity and IPv6 wins the happy eyeball race, and the link-local address is the only thing advertised.

Stripping the ifindex is technically invalid because now you can't know which link-local the request came in on, but most libraries are not designed to deal with link-local addresses unfortunately.

**Comment by mmerickel** [comment]
I'm fine with stripping off the `%` link local iface however I'm not going to do it because I don't know enough about it. If @bertjwregeer thinks we can do it then I'll be happy to accept the PR.

**Comment by bertjwregeer** [comment]
It's safe to do so for the use case of the debug toolbar. I wouldn't modify the vendored library, instead I'd strip it before it go there...

A link-local address without an interface index is not valid as an IP address though for connecting to/verifying whether one can route to it or not. The real fix would be to support interface indexes and store them, so that when displaying/presenting the IP address it correctly has the interface index associated with it. That's a longer discussion and one to be taken up with the stdlib.

**Comment by mmerickel** [comment]
Fixed by #307.
